app/controllers/api/vms_controller.rb

```diff
@@ -235,8 +235,12 @@ def vm_ident(vm)
     end
 
     def validate_vm_for_action(vm, action)
-      validation = vm.send("validate_#{action}")
-      action_result(validation[:available], validation[:message].to_s)
+      if vm.respond_to?("supports_#{action}?")
+        action_result(vm.public_send("supports_#{action}?"), vm.unsupported_reason(action.to_sym))
+      else
+        validation = vm.send("validate_#{action}")
+        action_result(validation[:available], validation[:message].to_s)
+      end
```

Comment:
@gauravaradhye I'm sensing a pattern here...perhaps this would be better as a helper method in the base controller?  @imtayadeway Thoughts?